### DISCUSSION TITLE
Tracking progress of a form upload

### DISCUSSION THREAD

**Discussion by el-garro** [aio-libs/aiohttp#6815] [Q&A]
Hi, I can´t figure out ho to make an async generator for a FormData  in order to track the upload progress, this is my working code:

```
async def sample_upload():
    with open("some_file.pdf", "rb") as file:
        data = aiohttp.FormData()
        data.add_field("title", "some_title")
        data.add_field("author", "some_authon")
        data.add_field("upload_file", file)

        async with aiohttp.ClientSession() as session:
            async with session.post(url="http://example.com/upload.php", data=data) as response:
                resp = await response.text()
```

If it was only the file, i could easily make generator yielding chunks, but i don't know how to make it work with a form. Thanks in advance.

**Comment by oliverpool** [comment]
Apparently someone found a solution using BufferedReader: https://stackoverflow.com/a/73728849/3207406

It counts the number of bytes read by aiohttp (not the numbers of bytes actually sent), but this might be enough for your usecase.
